lib/recog/nizer.rb

```diff
@@ -60,6 +132,48 @@ def self.multi_match(match_key, match_string)
   end
 
   #
+  # Search all fingerprint dbs and attempt to find **all** matching
+  # {Fingerprint fingerprint}s. Returns an array of all matching
+  # fingerprints or an empty array.
+  #
+  # @param match_string [String] Service banner to match
+  # @param [Hash] filters This hash contains filters used to limit the
+  #   results to just those from specific types of fingerprints.
+  #   The matched values are derived from the 'fingerprints' top level
+  #   element in the fingerprint DB XML or, in the case of 'protocol',
+  #   this value can be overridden at the individual fingerprint level.
+  # @option filters [String] :match_key Value from XML 'matches' or file name
+  # @option filters [String] :fp_type  Fingerprint type (service, os, etc.)
+  # @option filters [String] :protocol Protocol (ftp, smtp, etc.)
+  # @return [Array] Array of Fingerprint#match
```

Comment:
Minor, but mention the no-match edge-case which will result in this returning an empty array.  I prefer this method, FWIW, but it is common for some code to, say, return `nil` when there is no match.  Better to be explicit than leave it up in the air.